src/textual/widget.py

```diff
@@ -263,6 +276,8 @@ def _refresh_scrollbars(self) -> None:
             show_horizontal = True
         elif overflow_x == "auto":
             show_horizontal = self.virtual_size.width > width
+        if scrollbar_size_horizontal == 0:
```

Comment:
new specific case: a scrollbar size of zero should probably not result in any scrollbar display, right? :slightly_smiling_face: